### ISSUE TITLE
json function trimming decimal places

### ISSUE THREAD

**Issue by carlroutledge** [Azure/logicapps#467]
I'm pulling a message from the Azure Service bus, then using the base64tostring function inside the json function to get the content of the service bus into a format i can then use a Liquid Map to create my destination specific message.

What i've found is if the number coming in is something like 2500.0 it removes the .0 in the json function.  I did a bit of testing by using a compose just with the base64tostring function and the numbers are still correct (2500.0) but as soon as you then wrap that into the json function it is removed and rounded down i guess.

Basically my destination system requires the values (as the are monetary amounts) with 2 decimal places and i've tried everything to get this to happen but can't.

**Comment by AbodeSaafan** [comment]
Hey @carlroutledge 

Can you try using the 'decimal' expression? So it might look something like this: 

(code view) 
inputs": "@string(decimal('2500.0'))",

**Comment by carlroutledge** [comment]
Thanks for the response.  I may have tried a "| decimal" type command and it made no difference what so ever and it also needs to be as a number, so not surrounded by ".

I'll keep hunting but found no solution yet.

**Comment by carlroutledge** [comment]
Thanks for the reply.  I've stayed away from doing anything with the whole message in the logic apps.  I've just been retrieving the message from the Service Bus then using Liquid Map to convert into the correct message format for the destination.  If i do have to start doing something like you've suggested for each value that needs to be in that format i have a lot of logic apps to change and individual fields to check.

**Comment by AbodeSaafan** [comment]
I see, in that case I will tag in @xuehongg  @daviburg and @rarayudu who might know more about Liquid Map transformations

**Comment by carlroutledge** [comment]
Thanks.  It would be nice to fix in a liquid map but from my testing of Logic Apps it seems to be being removed by the json function in logic apps if it is .0 or .00.  I'm not saying even if that didn't happen in the json function the liquid map wouldn't then also remove it.

**Comment by daviburg** [comment]
How do you render the decimals in your Liquid template? By default Liquid trims the trailing zeroes of a float. Try the behavior online at http://dotliquidmarkup.org/try-online
with this sample template:
```
{% assign floatExplicitDecimals = -39.123 %}
{{floatExplicitDecimals}}
{% assign floatZeroesDecimals = -39.700 %}
{{floatZeroesDecimals}}
{% assign carlroutledge = 2500.0 %}
{{carlroutledge}}
```
I'm not sure if there is a liquid construct to express the number of decimals to show when rendering a float. Maybe the folks at https://gitter.im/dotliquid/dotliquid will have an idea.
